fix(OpenGL/RenderWindow): Stop adding bgImage to DOM

swederik opened the pull request:
Stop adding bgImage to DOM unless setBackgroundImage has been called.

Background images are necessary for remote rendering, but the `<img>` tag is currently being added to
the DOM unnecessarily, even if the user is not using remote rendering. This change prevents this
from occurring.

fix #1007 

This fix seems to be working for me.

jourdain commented:
So that means that if we add the image after binding the container to the RW, that image will never be added to the DOM.

I would be fine to have a specific flag `useBackgroundImage` that can be used to determine if the window should add that element or not. And its default can be false.

swederik commented:
Ah whoops, I didn't notice that. Let me fix that.

swederik commented:
Okay so here's the logic now:

`setContainer`:
- If bgImage exists in the previous container, remove it.
- If `useBackgroundImage` is true, add it to the new container

New publicAPI method: `setUseBackgroundImage`:
- Sets the value of `useBackgroundImage`.
- If it is true, and the bgImage does not already exist in the container, it is added.
- If it is false, and the bgImage exists in the container, it is removed.

So now you should be able to call `setUseBackgroundImage`, `setContainer`, and `setBackgroundImage` in any order.

An alternative to this would be to move the logic that is inside `setUseBackgroundImage` to `setBackgroundImage`. I figured that since the `setBackgroundImage` function is used for streaming in remote-rendered images, it would be called frequently. Adding an if statement to set the `useBackgroundImage` flag and conditionally add the DOM element that would not be important apart from the first time `setBackgroundImage` is called seemed wasteful. 

Let me know if you would prefer to ditch `setUseBackgroundImage` and move this inside `setBackgroundImage` instead.

Should I also add `useBackgroundImage` to macro.setGet call? Would that overwrite the definition of `setUseBackgroundImage`?

jourdain commented:
That's perfect. Maybe just adding the get method via the macro.

Also I would rather not do ```=== true / false``` but just use the actual variable with or without ```!``` in front of it.

swederik commented:
Wow I somehow didn't notice that `macro.get` existed. Added `useBackgroundImage` to `macro.get` and changed the if statements as suggested. PR should be good now.

Thanks for reviewing so quickly!

jourdain commented:
You are welcome! Thanks for doing the work and contributing it back...

jourdain commented:
:tada: This PR is included in version 8.3.11 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/vtk.js)
- [GitHub release](https://github.com/Kitware/vtk-js/releases/tag/v8.3.11)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
